verl/workers/rollout/async_server.py

```diff
@@ -0,0 +1,314 @@
+                request_id = request_id[len("chatcmpl-") :]
+                extra_headers["x-request-id"] = request_id
+
+            address = self.request_id_to_address[request_id]
+        else:
+            address = self.weighted_addresses[0][1]
+            self.weighted_addresses[0][0] += 1
+            heapq.heapreplace(self.weighted_addresses, self.weighted_addresses[0])
+
+            request_id = uuid4().hex
+            self.request_id_to_address[request_id] = address
+            chat_complete_request["extra_headers"]["x-request-id"] = request_id
+
+        completions, exception = None, None
+        try:
+            # TODO: OpenAI client uses httpx, seems to have performance issue in high concurrency requests.
+            completions = await self._chat_completions_openai(address, **chat_complete_request)
+        except Exception as e:
+            # Let user handle the exception
+            exception = e
+
+        await callback(completions, callback_additional_info, exception)
+
+    async def _chat_completions_openai(self, address: str, **chat_complete_request) -> ChatCompletion:
+        client = AsyncOpenAI(
+            base_url=f"http://{address}/v1",
+            api_key="token-abc123",
+        )
+        return await client.chat.completions.create(**chat_complete_request)
+
+    async def _chat_completions_aiohttp(self, address: str, **chat_complete_request) -> ChatCompletion:
+        try:
+            session = aiohttp.ClientSession()
+            async with session.post(
+                url=f"http://{address}/v1/chat/completions",
+                headers={"Authorization": "Bearer token-abc123"},
+                json=chat_complete_request,
+            ) as resp:
+                data = await resp.json()
+                return ChatCompletion(**data)
+        finally:
+            await session.close()
+
+    async def generate_sequences(self, prompts: DataProto, **sampling_params) -> DataProto:
+        raise NotImplementedError
+
+
+class AsyncLLMServerManager:
+    """AsyncLLMServerManager manage a group of vllm instances, i.e AsyncvLLMServer."""
+
+    def __init__(self, config: DictConfig, worker_group: RayWorkerGroup):
+        """Initialize AsyncLLMServerManager.
+
+        Args:
+            config: DictConfig, actor_rollout_ref config.
+            worker_group: RayWorkerGroup, worker group of AsyncActorRolloutRefWorker.
+        """
+        self.config = config
+        self.worker_group = worker_group
+
+        self.rollout_tp_size = self.config.rollout.tensor_model_parallel_size
+        self.rollout_dp_size = self.worker_group.world_size // self.rollout_tp_size
+
+        register_center = ray.get_actor(f"{self.worker_group.name_prefix}_register_center")
+        workers_info = ray.get(register_center.get_worker_info.remote())
+        assert len(workers_info) == self.worker_group.world_size
+
+        self.async_llm_workers = [None] * self.rollout_dp_size
```

Comment:
The `self.async_llm_workers` should be `self.async_llm_servers` as worker refers to GPU worker in verl